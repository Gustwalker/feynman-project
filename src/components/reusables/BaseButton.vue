<template>
  <a @click="$emit('click')" :class="`waves-effect waves-light btn ${buttonColor} ${textColor}`">
    <slot>

    </slot>
  </a>
</template>

<script>
export default {
  props: {
    buttonColor: {
      default () {
        return 'teal'
      },
      type: String 
    },
    textColor: {
      default () {
        return 'white-text'
      },
      type: String 
    }
  }
}
</script>