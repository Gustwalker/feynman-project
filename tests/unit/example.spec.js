import { shallowMount } from '@vue/test-utils'
// import Room from '@/views/Room.vue'
import BaseButton from '@/components/reusables/BaseButton.vue'

function sum(a, b) {
	return a + b
}

test('adds 1 + 2 to equal 3', () => {
	expect(sum(1, 2)).toBe(3)
})

describe('BaseButton', () => {
	it('renders props.msg when passed', () => {
		const wrapper = shallowMount(BaseButton)
		expect(wrapper.html()).toMatchSnapshot()
		// const msg = 'new message'
		// const wrapper = shallowMount(HelloWorld, {
		// 	propsData: { msg }
		// })
		// expect(wrapper.text()).toMatch(msg)
	})
})
