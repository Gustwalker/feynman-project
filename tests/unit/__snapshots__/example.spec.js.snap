// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButton renders props.msg when passed 1`] = `<a class="waves-effect waves-light btn teal white-text"></a>`;
